packages/api/test/fixtures/find-metrics.json

```diff
@@ -0,0 +1,15 @@
+{
```

Comment:
I wasted an hour whilst tired last night trying to figure out why my code worked in test and not on live and it was due to the mismatch of the response shape that you get from the graphql explorer / curl against fuana having a top level `data` property and our test code and db client hiding that. Could we make them all the same please? like have the fixtures match the response shape you get from the db, match the reponse shape you get from the db client?

cc @alanshaw

Reply:
If we do not return with `data`, the query simply does not return anything (this was the issue I was having yesterday).

```
const body = Query(
  Lambda(
    [],
    {
      data: {
        usersTotal: Count(Documents(Collection('User')))
        // ....
```

as opposed to:

```
const body = Query(
  Lambda(
    [],
    {
      usersTotal: Count(Documents(Collection('User')))
      // ....
```

With this in mind, I agree that we should return the same format always, but probably we will need to rely on data

Reply:
ok... a proposal
- the fixtures should include the `data` property. A fixture should match exactly what the db returns
- i'm going to PR an update to the api code where we call the db client to call response object `data` rather than `res`, so it's easier to guess that it has given you the `data` prop of the response rather than the full response with the `data` prop contained with in it.... which is what graphql-request does in all it's examples https://github.com/prisma-labs/graphql-request#quickstart

Reply:
I am a bit confused, we return with `data` in fauna but then the response that we get in the client has no data. I am receiving in the client the response just with `findMetrics`

Reply:
apparently only the status DB query returns data?

Reply:
https://github.com/web3-storage/web3.storage/blob/main/packages/db/fauna/resources/Function/findContentByCid.js#L25

I think what we want is to simply add a select to get the data from the Match here

Reply:
oh my, that could be the source of my confusion... is it a quirk of the `findContentByCid` function? why would the db client be stripping that part of the object???

Reply:
A [Match](https://docs.fauna.com/fauna/current/api/fql/functions/match?lang=shell#returns) returns a SetRef with data and then [Get](https://docs.fauna.com/fauna/current/api/fql/functions/get?lang=shell#returns) returns the first reference with data / ref too.

Reply:
Apparently `findUserByIssuer` also returns in the same format, but it is not being mocked. There seem to be the only two cases

Reply:
This is to do with returning a `Document` from FQL. A document is `{ ref, data, ttl, ... }`. When you return a _document_ from FQL, the value of `data` is what is returned in your GraphQL response.

This code is making a regular object _look_ like a document so that this happens - as vasco said, nothing is returned otherwise.

Note that `findContentByCid` is using `Get` which returns a `Document`.

---

Separately, when we use `Paginate` in FQL we return an array of documents. The directive `@resolver(paginated: true)` in GraphQL does a magic transform on that array of documents to turn it into `{ data, after, before }`, where `data` is the array of documents and `after`/`before` are auto generated cursor values to allow you to get the next/previous page of data.

e.g.

```gql
findUploadsCreatedAfter(since: Time!): [Upload!]! @resolver(paginated: true)
```

```gql
query {
  findUploadsCreatedAfter(since: "2021-07-08T15:13:08.318949Z" _cursor: null _size: 1000) {
    data {
      content {
        cid
      }
      user {
        issuer
      }
      created
    }
    # cursor value to be used as _cursor for the next page
    after
  }
}
```

A similar thing happens with `@relation` when the relation is a _many_ relationship:

```gql
batchEntries: [BatchEntry!]! @relation
```

```gql
  query {
      findContentByCid(cid: "bafy...") {
        dagSize
        batchEntries {
          data {
            dataModelSelector
            batch {
              cid
            }
          }
        }
      }
  }
```

Note that `BatchEntry` has a 1:1 relation with `Batch` so we don't end up with a `data` property in the selection. i.e. `batch { cid }` NOT `batch { data { cid } }`:

```gql
batch: Batch! @relation
```

Reply:
<img width="197" alt="Screenshot 2021-07-15 at 12 14 46" src="https://user-images.githubusercontent.com/152863/125779211-82d504b1-7a9b-4ecf-a8ff-d00a10d2fc18.png">

See what has happened here - `batchEntries` is actually a resolver function (with nullable params) that returns a page.